### ISSUE TITLE
Consider whether a11y daemons should be autostarted on `flatpak run

### ISSUE THREAD

**Issue by smspillaz** [flatpak/flatpak#1471]
Right now, when a flatpak is launched using `flatpak run` we create an a11y dbus proxy for it: https://github.com/flatpak/flatpak/blob/master/common/flatpak-run.c#L868 . This calls `GetAddress` on `org.a11y.Bus` in order to work out what the a11y bus is so that we can bind the bus on bwrap.

Recently I've been working on a server that runs as its own user through systemd socket activation and I needed access to a session bus, so I have been running flatpak through `dbus-run-session`. This works fine, except that when `flatpak run` starts, it calls that method which goes and autostarts the a11y bus and this takes upwards of 15 seconds on my system, probably due to the lack of available display servers inside the session.

Of course, my server doesn't care about a11y so all of this is essentially just wasted effort. I noticed that `flatpak_run_app` has a `FLATPAK_RUN_FLAG_NO_A11Y_BUS_PROXY` flag to prevent this from happening, and I've created a [pull request](https://github.com/endlessm/flatpak/pull/75) to make use of that from the CLI. However, @ramcq rightfully noted in a discussion that we had that what flatpak should *really* be doing is only starting and making available the proxy when the underlying application actually attempts to talk on those names(*). I'm not sure how that would currently be done with bwrap though, since it looks to me like things such as the `AT_SPI_BUS_ADDRESS` are set before the subprocess is spawned.

So, two questions come out of this:
1) Is there a path towards handling this properly?
2) Would flatpak upstream be interested in the patch to expose `FLATPAK_RUN_FLAG_NO_A11Y_BUS_PROXY` to the CLI in the meantime?

(*) There is another option, which is to work out whether the a11y bus is available before attempting to query it, but I'm not particularly fond of that because there's a hypothetical race condition where a flatpak gets started on session startup and the a11y bus names are not available yet.

**Comment by ramcq** [comment]
(There's also the possibility we tie the a11y proxying to another thing like whether any display sockets are proxied.)

**Comment by smspillaz** [comment]
The other option is to bring back `--no-desktop` and put it there, though I'm not sure if undeprecating things is the best idea.

**Comment by smspillaz** [comment]
I've opened #1486 (which was downstream change we decided not to have for now, but just wanted to continue the discussion upstream). If we're able to accept this change upstream or fix it in a better way, then we can drop a hack downstream where we have to start another instance of `at-spi-bus` for something that doesn't actually need it.
